server/src/game/index.ts

```diff
@@ -474,18 +570,22 @@ export class Game {
   }
 
   trainPlayerUnit(player: Player, coords: Coords, type: string): void {
-    const civID = player.civID;
-    this.world.map.trainUnitAt(coords, type, civID);
+    if (!this.hasStarted()) throw new GameNotStartedError();
+    const leaderID = player.leaderID;
+    const leader = this.leaders[leaderID];
+    this.world.map.trainUnitAt(coords, type, leader);
     this.sendUpdates();
   }
 
   settleCityAt(player: Player, coords: Coords, name: string): void {
-    const civID = player.civID;
+    if (!this.hasStarted()) throw new GameNotStartedError();
+    const leaderID = player.leaderID;
+    const leader = this.leaders[leaderID];
     const map = this.world.map;
 
     const unit = map.getTile(coords)?.unit;
-    if (unit?.type === 'settler' && unit?.civID === civID) {
-      const settlementSuccessful = map.settleCityAt(coords, name, civID, unit);
+    if (unit?.type === 'settler' && leader.controlsUnit(unit) && isCivDomain(unit.domainID)) {
```

Comment:
So Free Cities (or vassal cities) can't settle new cities right now? (I suppose that makes sense).

Reply:
Correct. I think (and feel free to disagree with me here) that cities that become completely independent and stable should just eventually turn into their own Civilization.